test/box-luatest/gh_8650_add_is_sync_option_test.lua

```diff
@@ -0,0 +1,319 @@
+        box.commit({is_sync = true})
+    end)
+
+    t.assert_error_msg_content_equals('Illegal parameters, is_sync ' ..
+                                      'can only be true',
+    function()
+        cg.master:exec(function()
+            box.begin({is_sync = false})
+                box.space.test:insert{5, 'is_sync = error'}
+            box.commit()
+            box.begin()
+                box.space.test:insert{6, 'is_sync = error'}
+            box.commit({is_sync = false})
+            box.begin({is_sync = "foo"})
+                box.space.test:insert{7, 'is_sync = error'}
+            box.commit()
+            box.begin()
+                box.space.test:insert{8, 'is_sync = error'}
+            box.commit({is_sync = "foo"})
+        end)
+    end)
+
+    t.assert_equals(cg.master:exec(function()
+        return box.space.test:select()
+    end), {{1, 'is_sync = false'}, {2, 'is_sync = true'},
+           {3, 'is_sync = true'}, {4, 'is_sync = true'}})
+
+    t.assert_equals(cg.replica:exec(function()
+        return box.space.test:select()
+    end), {{1, 'is_sync = false'}, {2, 'is_sync = true'},
+           {3, 'is_sync = true'}, {4, 'is_sync = true'}})
+
+
+    cg.replica:stop()
+
+    -- Async transaction
+    cg.master:exec(function()
+        box.begin()
+            box.space.test:insert{9, 'is_sync = false'}
+        box.commit()
+    end)
+
+    -- Sync transaction
+    t.assert_error_msg_content_equals('Quorum collection for a synchronous ' ..
+                                      'transaction is timed out',
+    function()
+        cg.master:exec(function()
+            box.begin({is_sync = true})
+                box.space.test:insert{10, 'is_sync = true'}
+            box.commit()
+        end)
+    end)
+
+    t.assert_equals(cg.master:exec(function()
+        return box.space.test:select()
+    end), {{1, 'is_sync = false'}, {2, 'is_sync = true'},
+           {3, 'is_sync = true'}, {4, 'is_sync = true'},
+           {9, 'is_sync = false'}})
+end
+
+g.test_box_atomic_is_sync = function(cg)
+    cg.master:exec(function()
+        box.schema.space.create('test')
+        box.space.test:create_index('pk')
+        box.ctl.promote()
+
+        box.atomic(function()
+            box.space.test:insert{1, 'is_sync = false'}
+            box.space.test:insert{2, 'is_sync = false'}
+        end)
+        box.atomic({is_sync = true}, function()
+            box.space.test:insert{3, 'is_sync = true'}
+            box.space.test:insert{4, 'is_sync = true'}
+        end)
+    end)
+
+    t.assert_error_msg_content_equals('Illegal parameters, is_sync ' ..
+                                      'can only be true',
+    function()
+        cg.master:exec(function()
+            box.atomic({is_sync = false}, function()
+                box.space.test:insert{5, 'is_sync = error'}
+                box.space.test:insert{6, 'is_sync = error'}
+            end)
+            box.atomic({is_sync = "foo"}, function()
```

Comment:
Same as above, you should break these calls into two functions. Otherwise only the first one is tested.

Reply:
Fixed